From d42e5a51679297ddd9b7b32626258be4ceb02be0 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Fri, 12 Aug 2011 09:59:55 +0200
Subject: [PATCH] patch: crop-area

Squashed commit of the following:

commit bc74edae3dd5b3903e191bbaf816130b42cd9a33
Author: udifuchs <udifuchs>
Date:   Fri Aug 12 03:15:05 2011 +0000

    Reset crop data between image loading (in RC).
    (cherry picked from commit fb61698d7fce847878a2b02f43e1e51886b34e11)
---
 ufraw_preview.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/ufraw_preview.c b/ufraw_preview.c
index c0cfefa..a48c1ef 100644
--- a/ufraw_preview.c
+++ b/ufraw_preview.c
@@ -3453,12 +3453,14 @@ static void delete_from_list(GtkWidget *widget, gpointer user_data)
     gtk_dialog_response(dialog, GTK_RESPONSE_APPLY);
 }
 
-// Duplicate CFG into RC, except for ufobject which is copied elegantly
+// Duplicate CFG into RC, except for transform data, which is reset and
+// ufobject which is copied elegantly
 static void copy_conf_to_rc(preview_data *data)
 {
     UFObject *tmp = RC->ufobject;
     *RC = *data->UF->conf;
     RC->ufobject = tmp;
+    conf_copy_transform(RC, &conf_default);
     UFObject *image = ufgroup_element(RC->ufobject, ufRawImage);
     ufobject_copy(image, data->UF->conf->ufobject);
 }
-- 
1.7.6

